compiler/semtypes.nim

```diff
@@ -1403,7 +1403,7 @@ proc semGeneric(c: PContext, n: PNode, s: PSym, prev: PType): PType =
   elif t.kind != tyGenericBody:
     # we likely got code of the form TypeA[TypeB] where TypeA is
     # not generic.
-    localError(c.config, n.info, errNoGenericParamsAllowedForX % s.name.s)
+    localError(c.config, n.info, errNoGenericParamsAllowedForX % $(s.name.s, t.kind))
```

Comment:
Seriously? ``tyFoobar`` in error messages?

Reply:
fixed via a reusable `toHumanStr`; note that `typeToString()` is not enough (doesn't help debugging what went wrong in many cases: the type name doesn't tell you the type kind).
now you'll get (in the other location under evalTypeTrait, not here) error messages like:
```nim
Error: expected genericInst, got: sequence of type seq[int]
```

Reply:
Still not good enough. "expected genericInst", who is supposed to understand that? And how exactly is ``type seq[int]`` not an instantiation?

Reply:
* I've removed this change here to unblock this PR although IMO more relevant debug info is better and helps anyone with some familiarity with compiler internals (s.name.s doesn't show the typeKind and isn't as useful). And I've changed the other message to show error messages like this:
```nim
 Error: expected generic type, got: type Bar of kind object
```
which IMO is much more helpful than:
```nim
 Error: expected generic type, got: type Bar
```

> And how exactly is type seq[int] not an instantiation?

I don't know how to support `tySequence` so I'm leaving this part for future PR; 
in particular this does not work:
```nim
var resType = newType(tySequence, operand.owner)
result = toNode(resType, traitCall.info)
```
This PR is a strict improvement over status quo, where genericHead did not work at all, now it works for generic objects (but not yet `seq[T]`) (in addition to fixing the other 2 issues)